experimenter/experimenter/experiments/models.py

```diff
@@ -1285,30 +1287,101 @@ def timeline(self):
+            analysis_basis, segment, reference_branch, window
+        )
+
+        metrics_metadata = (
+            self.results_data.get("v3", {}).get("metadata", {}).get("metrics")
+            if self.results_data
+            else {}
+        ) or {}
+        all_outcome_metric_slugs = []
+        for slug in self.primary_outcomes + self.secondary_outcomes:
+            outcome = Outcomes.get_by_slug_and_application(slug, self.application)
+            metrics = outcome.metrics if outcome else []
+            outcomes = []
+
+            for metric in metrics:
+                formatted_metric = {
+                    "slug": metric.slug,
+                    "description": metric.description
+                    if metric.description
+                    else metrics_metadata.get(metric.slug, {}).get("description", ""),
+                    "group": "other_metrics",
+                    "friendly_name": metric.friendly_name
+                    if metric.friendly_name
+                    else metrics_metadata.get(metric.slug, {}).get(
+                        "friendly_name", metric.slug
+                    ),
+                }
+                if formatted_metric not in outcomes:
+                    outcomes.append(formatted_metric)
+                    all_outcome_metric_slugs.append(metric.slug)
+
+            outcomes.sort(key=lambda m: m["friendly_name"])
+            metric_areas[outcome.friendly_name if outcome else slug] = outcomes
+
+        metric_areas[NimbusUIConstants.OTHER_METRICS_AREA] = self.get_default_metrics(
+            all_outcome_metric_slugs
+        )
+
+        window_results = self.get_window_results(analysis_basis, segment, window)
+
+        def metric_is_notable(slug, group):
+            for branch_data in window_results.values():
+                metric_data = (
+                    branch_data.get("branch_data", {}).get(group, {}).get(slug, {})
+                )
+                for branch_significance in metric_data.get("significance", {}).values():
+                    if (
+                        "positive" in branch_significance.get(window, {}).values()
+                        or "negative" in branch_significance.get(window, {}).values()
+                    ):
+                        return True
+            return False
 
-        # TODO(13721): check all metrics with data and add their areas to metric_areas
-        # if not already present
+        for metrics in metric_areas.values():
+            for metric in metrics:
+                if (
+                    metric_is_notable(metric["slug"], metric["group"])
+                    and metric not in notable_metrics
+                ):
+                    notable_metrics.append(metric)
 
+        notable_metrics.sort(key=lambda m: m["friendly_name"])
         return metric_areas
 
-    @property
-    def default_metrics(self):
+    def get_default_metrics(self, exclude_slugs=None):
```

Comment:
Without passing any "exclude" slugs, this will return more than just the default metrics, no? Should it be called something like `get_other_metrics`? Or, depending on how the new UI groups metrics, maybe this could take the UI's group name (maybe an enum in `NimbusUIConstants` defines the possible areas?) and return the metrics for that group?

And this is more for my own curiosity, but it strikes me as a bit odd to exclude metrics as opposed to saying which ones you do want. What's the reasoning there?

Reply:
Also consider naming this something that is clear that it returns metadata and not results.